apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -720,27 +826,51 @@ static void tc_fs_vfs_rewinddir(void)
 static void tc_fs_vfs_seekdir(void)
 {
 	int ret;
-	DIR *dirp;
+	DIR *dir;
 	struct dirent *dirent;
 	off_t offset;
 	char filename[1];
 
-	dirp = opendir(VFS_FOLDER_PATH);
-	TC_ASSERT_NEQ("opendir", dirp, NULL);
+	dir = opendir(VFS_FOLDER_PATH);
+	TC_ASSERT_NEQ("opendir", dir, NULL);
 
 	offset = 2;
-	seekdir(dirp, offset);
-	TC_ASSERT_NEQ_CLEANUP("seekdir", dirp, NULL, closedir(dirp));
-	dirent = readdir(dirp);
-	TC_ASSERT_NEQ_CLEANUP("readdir", dirent, NULL, closedir(dirp));
-	TC_ASSERT_EQ_CLEANUP("readdir", dirent->d_type, DTYPE_DIRECTORY, closedir(dirp));
+	seekdir(dir, offset);
+	TC_ASSERT_NEQ_CLEANUP("seekdir", dir, NULL, closedir(dir));
+	dirent = readdir(dir);
+	TC_ASSERT_NEQ_CLEANUP("readdir", dirent, NULL, closedir(dir));
+	TC_ASSERT_EQ_CLEANUP("readdir", dirent->d_type, DTYPE_DIRECTORY, closedir(dir));
 
-	ret = closedir(dirp);
+	ret = closedir(dir);
 	TC_ASSERT_EQ("closedir", ret, OK);
 
 	itoa((int)offset, filename, 10);
 	TC_ASSERT_EQ("readdir", strncmp(dirent->d_name, filename, 1), 0);
 
+	/* For Negative offset in seekmountdir operations */
+	dir = opendir(VFS_FOLDER_PATH);
+	TC_ASSERT_NEQ("opendir", dir, NULL);
+
+	offset = -2;
+	seekdir(dir, offset);
+	TC_ASSERT_NEQ_CLEANUP("seekdir", dir, NULL, closedir(dir));
+
+	ret = closedir(dir);
+	TC_ASSERT_EQ("closedir", ret, OK);
+	/* for pseudo dir operations */
+
+	dir = opendir(ROOT_PATH);
+	TC_ASSERT_NEQ("opendir", dir, NULL);
+
+	offset = 2;
```

Comment:
can remove hard coded value and replace by SEEK_END

Reply:
yes, It is added in this commit 446ba7fcb5b5f7bb23e610ecf84afb1b2922d430